File: iniciante/2162.py
def main():
    qnt = int(input())
    montanhas = list(map(int, input().split()))

    print(test(montanhas))


def test(montanhas):
    tam = len(montanhas)
    if tam == 1:
        return 1
    elif tam == 2:
        return 1 if montanhas[0] != montanhas[1] else 0
    
    maior = montanhas[0] > montanhas[1]
    for i in range(1, tam - 1):
        if maior and montanhas[i] >= montanhas[i + 1]:
            return 0
        if not maior and montanhas[i] <= montanhas[i + 1]:
            return 0
        maior = not maior

    return 1


if __name__ == "__main__":
    main()

File: iniciante/2162.cpp
#include <iostream>
using namespace std;

int vertificar(int vetor[], int qnt)
{
    if(qnt == 1)
        return 1;
    else if(qnt == 2)
    {
        if (vetor[0] !=  vetor[1])
            return 1;
        else
            return 0;
    }
    else
    {   
        bool maior = vetor[0] >  vetor[1];
        for(int i = 1; i < qnt - 1; i ++)
        {
            if(maior && vetor[i] >= vetor[i + 1])
                return 0;
            else if (!maior && vetor[i] <= vetor[i + 1])
                return 0;
            
            maior = !maior;
        }       
        return 1;
    }

}

int main()
{
    int qnt;
    cin >> qnt;
    int montanhas[qnt + 10];
    for(int i = 0; i < qnt; i ++)
        cin >> montanhas[i];
    
    cout << vertificar(montanhas, qnt) << endl;

    return 0;
}